griptape/drivers/vector/mongodb_vector_store_driver.py

```diff
@@ -80,11 +80,58 @@ def load_entries(
             )
 
     def query(
-        self,
-        query: str,
-        count: Optional[int] = None,
-        namespace: Optional[str] = None,
-        include_vectors: bool = False,
-        **kwargs
+            self,
+            query: str,
+            count: Optional[int] = None,
+            namespace: Optional[str] = None,
+            include_vectors: bool = False,
+            offset: Optional[int] = 0,
+            **kwargs
     ) -> list[BaseVectorStoreDriver.QueryResult]:
-        raise NotImplementedError("Vector search not supported in MongoDB")
+        collection = self.get_collection()
+
+        # Using the embedding driver to convert the query string into a vector
+        vector = self.embedding_driver.embed_string(query)
+        print("Vector for query:", vector)
+
+        knn_k = count if count else 10
+        pipeline = [
+            {
+                "$search": {
+                    "index": "knn",  # Use the name of the k-NN index you created
+                    "knnBeta": {
+                        "vector": vector,
+                        "path": "vector",
+                        "k": knn_k + offset
+                    }
+                }
+            },
+            {
+                "$project": {
+                    "_id": 1,
+                    "vector": 1,
+                    "namespace": 1,
+                    "meta": 1
+                }
+            },
+            {"$skip": offset},
+            {"$limit": knn_k}
+        ]
+
+        cursor = collection.aggregate(pipeline)
+        #print("Pipeline:", pipeline)
+        #print("Cursor:", list(cursor))
+        list_cursor = list(cursor)
+        for doc in list_cursor:
+            print(doc)
+        results = [
+            BaseVectorStoreDriver.QueryResult(
+                vector=doc["vector"] if include_vectors else None,
+                score=None,  # Score is not directly provided by knnBeta
```

Comment:
I think [it is](https://www.mongodb.com/docs/atlas/atlas-search/knn-beta/#scoring) :)